### PR TITLE
feat: add nuxt-session

### PR DESCRIPTION
**Pull request by BracketJohn**
Add nuxt-session to the modules overview.

### CODE REVIEW

**Comment by Atinux** [comment]
Thanks! I am not sure about using the sidebase logo for this since it aims to be generic right?

**Comment by BracketJohn** [comment]
> Thanks! 

Sure 🎊 


>  I am not sure about using the sidebase logo for this since it aims to be generic right? 

I agree - I just wanted to have some sort of logo and didn't have one handy :D 

Let me see if I can find a quick solution.

File: modules/session.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: session
+description: 'Session middleware to persist data across multiple requests, supports many backends via unjs/unstorage: memory, redis, fs, ...'
+repo: sidebase/nuxt-session
+npm: '@sidebase/nuxt-session'
+icon: session.svg
+github: https://github.com/sidebase/nuxt-session
+website: https://github.com/sidebase/nuxt-session
+learn_more: https://github.com/sidebase
+category: Request
```
Comment (Atinux) [inline modules/session.yml:9]:
```suggestion
category: Security
```

**Comment by Atinux** [comment]
I think you can remove it, I suggested to move to "Security" as category, so this will be using this logo by default:
CleanShot 2022-10-12 at 17 57 06

**Comment by BracketJohn** [comment]
Added a new logo + changed the category @Atinux - what do you think about this one? ☺️

**Comment by BracketJohn** [comment]
@Atinux I had to fix the icon location but now it should be good to go 💯

**Comment by Atinux** [comment]
Do you mind improving the playground of your module to show and example of it please?

Also it would be nice to add the badges on the readme to also show the npm version and license.

**Comment by BracketJohn** [comment]
Sure thing - I'll do that tomorrow though, tonight I'm already off my PC

See you tomorrow 🤙

**Comment by BracketJohn** [comment]
> Do you mind improving the playground of your module to show and example of it please?

done, added a playground + screencap, check it out here: https://github.com/sidebase/nuxt-session#playground

Sneak peak:
image

> Also it would be nice to add the badges on the readme to also show the npm version and license.

Done, badges:
image

url: https://github.com/sidebase/nuxt-session#readme

---

@Atinux, feel free to check out the playground and see if it works. Otherwise let me know if I should change anything further or whether we're ready to merge 🚀

**Comment by Atinux** [comment]
Checking the playground https://stackblitz.com/github/sidebase/nuxt-session?file=README.md

Increasing session does not work.

**Comment by BracketJohn** [comment]
I checked it out - _I think_ it has something to do with webcontainers that stackblitz uses + how cookies behave in them.

E.g., when I want to see all cookies on an `event` I get an empty object every time:
```ts
console.log(parseCookies(event)) // -> {}
```

 I tested with firefox and chrome locally and there it worked - I assume it still actually works for all local & deployed projects,. I'll dig into whether stackblitz-webcontainer-cookie-problems are a thing. If this maybe already rings a bell for you, I'd be happy for input.

**Comment by BracketJohn** [comment]
I created a minimal reproduction to show how cookies do not correctly work within stackblitz. This repro does not contain any `nuxt-session` code, it's a minimal middleware with 10 LoCS. Here's the url: https://stackblitz.com/edit/nuxt-session-cookie-repro?file=playground/app.vue

To test it:
1. Launch the app,
2. after it starts, click the button
3. Have a look at the server-logs

In stackblitz, we get:
```sh
1. cookies start: {}
2. setting cookie hello
3. cookies end ("hello" should be set):  {}
```

When testing this locally, I get:
```
1. cookies start: {
  hello: 'Thu Oct 13 2022 16:27:42 GMT+0200 (Central European Summer Time)'
}
2. setting cookie hello
3. cookies end ("hello" should be set):  {
  hello: 'Thu Oct 13 2022 16:27:42 GMT+0200 (Central European Summer Time)'
}
```

This uses RC.11.  

Assuming that this means that an underlying problem P exists: "coolies don't work in stackblitz" / "h3 cookie utils do not work in stackblitz":
1. I think if it's a goal that `nuxt-session` works in stackblitz, we have to resolve the underlying problem P in stackblitz first
2. If resolving the underlying problem is a no goal / this is part of a bigger problem: Can we merge this PR anyways, as the problem is not one of `nuxt-session` and `nuxt-session` works in local tests (of course this statement may still be challenged!)?

I may be missing something here, but I hope that the reproduction shows that the problem is not actually related to how `nuxt-session` is implemented.

**Comment by danielroe** [comment]
Not checked in great detail, but when you are setting a cookie on an event, it sets it on the response. When you are _reading_ cookies on the event, it reads them from the request.

I'm not sure how this was working locally.

**Comment by BracketJohn** [comment]
But then on the next request, the cookie that was set on the response should also be sent, right? That's what I was trying to say with:

>  (setCookie sets the cookie on the response, then the next request has it attached)

**Comment by danielroe** [comment]
Sorry, understood!

**Comment by BracketJohn** [comment]
No worries! 

Looking further, [even after following stacklitz guidance on browser config to allow all cookies](https://developer.stackblitz.com/platform/webcontainers/browser-config) there's nothing (e.g., no `Set-Cookie` header) coming back in the response on stackblitz:
image

Locally however, chrome even offers a "cookies" tab:
image

**Comment by danielroe** [comment]
It seems stackblitz doesn't currently support setting cookies.

**Comment by BracketJohn** [comment]
I see, that's a bummer 😢 

Do you think we can progress with this PR anyways?

**Comment by BracketJohn** [comment]
Hey @Atinux, @danielroe - can you guys have another look here? I've added the badges + debugged stackblitz together with you, so I think at the moment nothing else is blocking this.

We'd love to get this module out there + support it in the longterm. We already have ideas like enabling session transmission via a header token instead of a cookie, auto adding endpoints for session management via a config flag, adding conposables to fetch the session client side, ... 

I think adding the module to the modules page will be a good step to support this endeavor 🙌

Let me know what remains if you can, otherwise feel free to reach out to me via Twitter / Discord if you want to discuss (:

**Pull request merged**